chain/client/src/client.rs

```diff
@@ -2348,6 +2348,14 @@ impl Client {
         Ok(false)
     }
 
+    pub fn catchup_shards(&self) -> Result<Vec<ShardId>, Error> {
+        let mut shards = HashSet::new();
+        for (_sync_hash, state_sync_info) in self.chain.store().iterate_state_sync_infos()? {
+            shards.extend(state_sync_info.shards.iter().map(|tuple| tuple.0));
```

Comment:
```suggestion
            shards.extend(state_sync_info.shards.iter().map(|shard_info| shard_info.0));
```

Reply:
ya agreed that's better, I just copy pasted this code from below :)